agent/worker.go

```diff
@@ -484,3 +489,36 @@ func (w *worker) Subscribe(ctx context.Context, subscription *api.SubscriptionMe
 		}
 	}
 }
+
+// Leave removes all assignments
+func (w *worker) Leave(ctx context.Context) error {
+	var err error
+	w.Close()
```

Comment:
It shouldn't really be possible to call `Close` then `Assign`. `Close` shuts down the worker for future activity.